examples/self_supervised_learning/losses/_wav2vec2_loss.py

```diff
@@ -0,0 +1,84 @@
+from typing import Optional, Tuple
+
+import torch
+import torch.nn.functional as F
+from torch import Tensor
+
+
+def compute_contrastive_loss(
+    x: Tensor,
+    mask_indices: Tensor,
+    targets: Tensor,
+    neg_is_pos: Tensor,
+    reduce: bool,
+    logit_temp: float = 0.1):
+    """
+    Computes the contrastive loss used in Wav2Vec2 loss function.
+
+    Args:
+        x (Tensor): Input embeddings of shape (batch_size, sequence_length, hidden_size)
+        mask_indices (Tensor): Indices to mask negative samples
+        targets (Tensor): Labels indicating positive samples of shape (batch_size, sequence_length, hidden_size).
```

Comment:
The shape of target should be `(num_negative + 1, batch, sequence_length, hidden_size)`, right?